lib/webistrano/deployer.rb

```diff
@@ -270,33 +226,57 @@ def self.type_cast(val)
         end
       end
     end
-    
+
     def self.cvs_root_defintion?(val)
       val.index(':') == 0 && val.scan(":").size > 1
     end
-    
-    # override in order to use DB logger 
+
+    # override in order to use DB logger
     def handle_error(error) #:nodoc:
       case error
       when Net::SSH::AuthenticationFailed
         logger.important "authentication failed for `#{error.message}'"
-      when Capistrano::Error
-        logger.important(error.message)
-      else 
+      # when Capistrano::Error
+      #   logger.important(error.message)
+      else
         # we did not expect this error, so log the trace
         logger.important(error.message + "\n" + error.backtrace.join("\n"))
       end
     end
-    
+
     # returns a list of all tasks defined for this deployer
     def list_tasks
-      config = instantiate_configuration
-      config.load 'deploy'
-      
-      set_up_config(config)
-      
-      config.task_list(:all)
+      [{:name => "deploy:rollback", :description => nil}, {:name => "deploy:cleanup", :description => nil}]
+    end
+
+    def find_or_create_project_dir(project)
+      FileUtils.mkdir_p(rooted('capsize_projects')) unless Dir.exists?(rooted('capsize_projects'))
+      FileUtils.mkdir_p(rooted("capsize_projects/#{project}")) unless Dir.exists?(rooted("/capsize_projects/#{project}"))
```

Comment:
Two things:

1. DRY. Assign the rooted path name to a variable, so you know it is (and stays) consistent. In general, think about how things can get broken when they change. If you ever go, "Well, if they change this, they'll also have to change that...", then you have bad code.
2. I believe the second case line will handle the case of the first. Can you write a test demonstrating otherwise? This is where TDD—especially for utility methods like this—is very useful. It will actually save you code.